packages/@expo/config-types/src/ExpoConfig.ts

```diff
@@ -176,6 +176,10 @@ export interface ExpoConfig {
      * By default, expo-updates will check for updates every time the app is loaded. Set this to `ON_ERROR_RECOVERY` to disable automatic checking unless recovering from an error. Set this to `NEVER` to disable automatic checking. Valid values: `ON_LOAD` (default value), `ON_ERROR_RECOVERY`, `WIFI_ONLY`, `NEVER`
      */
     checkAutomatically?: 'ON_ERROR_RECOVERY' | 'ON_LOAD' | 'WIFI_ONLY' | 'NEVER';
+    /**
+     * Whether to use embedded updates. Defaults to true. If set to false, an update will be fetched at launch. Ensure that `updates.checkAutomatically` is enabled if you disable embedded updates, and increase the `updates.fallbackToCacheTimeout`. You should not use this field in a production deployment.
```

Comment:
Given this description, we need some validation code that either throws, or prints a warning, if `checkAutomatically` is not set to `ALWAYS`, or `fallbackToCacheTimeout` is 0.

Reply:
i have a diff for this locally but i'm not sure it makes sense to do here and not for other options. for example, we support `checkAutomatically: 'NEVER'` and `fallbackToCacheTimeout: 1000` which are technically a combination that makes no sense. it's not clear to me that validation like this should be done at the prebuild stage

Reply:
i pushed the diff anyways with a comment about whether this is the right place for this functionality or not